API_AND_DATABASE_DESIGN.md

```diff
@@ -0,0 +1,1738 @@
+- Weekly reflections (Sunday) require structured `content` field
+- Daily reflections can use simple `text` field
+- `isShared` makes reflection visible to mentor
+- `aiInsights` generated via background job after reflection submission
+
+---
+
+### 10. SessionFeedback (Detailed Session Feedback)
+
+**Purpose**: Detailed session feedback form with growth areas and improvement suggestions
+
+**DynamoDB Table**: `SessionFeedbacks`
+
+- **PK**: `sessionId` (String)
+- **SK**: (None - one feedback per session)
+
+- **Attributes**:
+  - `feedbackId` (String) - UUID (for GraphQL ID field)
+  - `menteeId` (String) - Mentee user ID
+  - `mentorId` (String) - Mentor user ID
+  - `rating` (Number) - Rating (1-5)
+  - `growthArea` (String) - Primary growth area discussed
+  - `growthIdea` (String) - Actionable growth idea
+  - `whatWentWell` (String) - Positive feedback
+  - `whatToImprove` (String) - Improvement suggestions
+  - `additionalComments` (String) - Additional comments
+  - `isAnonymous` (Boolean) - Hide mentee identity
+  - `createdAt` (String) - ISO timestamp
+  - `updatedAt` (String) - ISO timestamp
+
+**GraphQL Type**:
+
+```graphql
+type SessionFeedback {
+  feedbackId: ID!
+  session: Session!
+  mentee: User!
+  mentor: User!
+  rating: Int!
+  growthArea: String!
+  growthIdea: String!
+  whatWentWell: String!
+  whatToImprove: String!
+  additionalComments: String
+  isAnonymous: Boolean!
+  createdAt: String!
+  updatedAt: String!
+}
+```
+
+**Access Patterns**:
+- Get feedback by session: Query by `PK = sessionId`
+- List feedbacks for mentor: Query GSI by `mentorId`
+- List feedbacks for mentee: Query GSI by `menteeId`
+
+**Business Rules**:
+- One feedback per session (enforced by PK)
+- Can only be submitted 24 hours after session completion
+- Expires 7 days after session completion
+- `isAnonymous` hides mentee identity from mentor
+
+---
+
+### 11. Review (General Reviews)
+
+**Purpose**: General reviews for mentors/mentees (not session-specific)
+
+**DynamoDB Table**: `Reviews`
+
+- **PK**: `reviewId` (String) - UUID
+- **SK**: (None - single item per review)
+
+- **Attributes**:
+  - `authorId` (String) - User ID of reviewer
+  - `targetId` (String) - User ID being reviewed
+  - `rating` (Number) - Rating (1-5)
+  - `comment` (String) - Review comment
+  - `type` (String) - `SESSION_FEEDBACK`, `GENERAL`
+  - `createdAt` (String) - ISO timestamp
+  - `updatedAt` (String) - ISO timestamp
+
+**GraphQL Type**:
+
+```graphql
+type Review {
+  reviewId: ID!
+  author: User!
+  target: User!
+  rating: Int!
+  comment: String
+  type: ReviewType!
+  createdAt: String!
+  updatedAt: String!
+}
+
+enum ReviewType {
+  SESSION_FEEDBACK
+  GENERAL
+}
+```
+
+**Access Patterns**:
+- Get review by ID: Query by `PK = reviewId`
+- List reviews for target: Query GSI by `targetId`, sort by `createdAt`
+- List reviews by author: Query GSI by `authorId`
+
+---
+
+### 12. Event (Community Events)
```

Comment:
I feel like we have not really discussed the uses of this yet, so I am going to skip this for now.